### PR TITLE
feat(layout): convert main navigation from sidebar to toolbar

### PR DESCRIPTION
**Pull request by shanberg**
[Demo](shanberg.github.io/athens)

Reformats main navigation into a toolbar at the top of the page. Toolbar allows for more general navigation while the sidebar is closed, and takes less total pixel area than the sidebar.

Screen Shot 2020-07-11 at 9 39 56 PM

### CODE REVIEW

File: src/cljs/athens/devcards/app_toolbar.cljs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    [:header (use-style app-header-style)
+     [:div (use-style app-header-control-section-style)
+      [button {:active (when @left-open? true)
+               :label [:> mui-icons/Menu] :on-click-fn #(dispatch [:left-sidebar/toggle])}]
+      [separator]
+      [button {:on-click-fn #(navigate :home)
+               :label [:> mui-icons/ChevronLeft]}]
+      [button {:on-click-fn #(navigate :home)
+               :label [:> mui-icons/ChevronRight]}]
```
Comment (tangjeff0) [inline src/cljs/athens/devcards/app_toolbar.cljs:85]: comment or disable for now

File: src/cljs/athens/devcards/app_toolbar.cljs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+(defn app-toolbar
+  []
+  (let [left-open? (subscribe [:left-sidebar/open])
+        right-open? (subscribe [:right-sidebar/open])
+        current-route (subscribe [:current-route])
+        route-name (-> @current-route :data :name)]
+
+    [:header (use-style app-header-style)
+     [:div (use-style app-header-control-section-style)
+      [button {:active (when @left-open? true)
```
[the remark this answers is not in the document]
Comment (tangjeff0) [reply src/cljs/athens/devcards/app_toolbar.cljs:79]: few other spots too

File: src/cljs/athens/views.cljs [hunk 3]
```diff
@@ -22,10 +23,11 @@
 (def app-wrapper-style
   {:display "grid"
    :grid-template-areas
-   "'left-sidebar main-content secondary-content'
+   "'app-header app-header app-header'
+    'left-sidebar main-content secondary-content'
    'devtool devtool devtool'"
```
Comment (tangjeff0) [inline src/cljs/athens/views.cljs:28]: Might be a good place to leverage garden's vector format

File: src/cljs/athens/views.cljs [hunk 4]
```diff
@@ -111,7 +114,8 @@
          (if @loading
            [initial-spinner-component]
            [:div (use-style app-wrapper-style)
-            [left-sidebar]
+            [app-toolbar route-name]
+            [left-sidebar route-name]
```
Comment (tangjeff0) [inline src/cljs/athens/views.cljs:118]: remove `route-name` param

File: src/cljs/athens/views.cljs (hunk 3 above)
Comment (shanberg) [reply src/cljs/athens/views.cljs:28]:
```clojure
[["'app-header app-header app-header'"
  "'left-sidebar main-content secondary-content'"
  "'devtool devtool devtool'"]]
```
vs 

```clojure
"'app-header app-header app-header'
 'left-sidebar main-content secondary-content'
 'devtool devtool devtool'"
```

Assuming I'm doing this correctly, given the formatting of the required CSS output, I think the regular string is simpler.

Comment (tangjeff0) [reply src/cljs/athens/views.cljs:28]: Thought we could do something else, but the exepected CSS is a little trickier because it expects a list of strings

**Pull request merged**